app/bundles/LeadBundle/Field/Command/CreateCustomFieldCommand.php

```diff
@@ -61,66 +65,100 @@ protected function execute(InputInterface $input, OutputInterface $output): int
+        if ($all && $leadFieldId) {
+            $output->writeln('<error>'.$this->translator->trans('mautic.lead.field.all_option_conflict').'</error>');
 
-        if (!$this->checkRunStatus($input, $output, $moderationKey)) {
-            return \Symfony\Component\Console\Command\Command::SUCCESS;
+            return Command::FAILURE;
         }
+
+        if ($all) {
+            return $this->addAllMissingColumns($input, $output) ? Command::SUCCESS : Command::FAILURE;
+        }
+
         if (!$leadFieldId) {
-            $leadField = $this->leadFieldRepository->getFieldThatIsMissingColumn();
+            @trigger_error('Must pass an id with the --id flag or use the --all flag. Future versions will use --all as the default behaviour', E_USER_DEPRECATED);
 
-            if (!$leadField) {
-                $output->writeln('<info>'.$this->translator->trans('mautic.lead.field.all_fields_have_columns').'</info>');
+            return $this->findAndAddColumn($input, $output) ? Command::SUCCESS : Command::FAILURE;
+        }
+
+        return $this->addColumn($leadFieldId, $userId, $input, $output) ? Command::SUCCESS : Command::FAILURE;
+    }
 
-                return \Symfony\Component\Console\Command\Command::SUCCESS;
+    private function addAllMissingColumns(InputInterface $input, OutputInterface $output): bool
+    {
+        $hasNoErrors = true;
+        while ($leadField = $this->leadFieldRepository->getFieldThatIsMissingColumn()) {
+            if (!$this->addColumn($leadField->getId(), $leadField->getCreatedBy(), $input, $output)) {
+                $hasNoErrors = false;
             }
+        }
+
+        return $hasNoErrors;
+    }
+
+    private function findAndAddColumn(InputInterface $input, OutputInterface $output): bool
+    {
+        $leadField = $this->leadFieldRepository->getFieldThatIsMissingColumn();
+
+        if (!$leadField) {
+            $output->writeln('<info>'.$this->translator->trans('mautic.lead.field.all_fields_have_columns').'</info>');
 
-            $leadFieldId = $leadField->getId();
-            $userId      = $leadField->getCreatedBy();
+            return true;
+        }
+
+        return $this->addColumn($leadField->getId(), $leadField->getCreatedBy(), $input, $output);
+    }
+
+    private function addColumn(int $leadFieldId, ?int $userId, InputInterface $input, OutputInterface $output): bool
+    {
+        $moderationKey = sprintf('%s-%s-%s', self::COMMAND_NAME, $leadFieldId, $userId);
+
+        if (!$this->checkRunStatus($input, $output, $moderationKey)) {
+            return true;
         }
 
         try {
             $this->backgroundService->addColumn($leadFieldId, $userId);
         } catch (LeadFieldWasNotFoundException $e) {
             $output->writeln('<error>'.$this->translator->trans('mautic.lead.field.notfound').'</error>');
 
-            return \Symfony\Component\Console\Command\Command::FAILURE;
+            return false;
```

Comment:
Commands must return `int`. I don't understand why this was changed on many places.

Reply:
@diaboloshogunate this is still an issue.

Reply:
Oh, now I noticed it's an if statement and not just returning the bool value. I'm not sure it is an improvement in readability over the previous state. Maybe it's just my dyslexia.

Reply:
I removed the if statement so it returns the int value for better readability